CourseStream/CourseStream.Data/CourseStreamRepository.cs

```diff
@@ -21,4 +21,22 @@ public async Task<List<CourseStream>> ListAsync()
     {
         return await _context.CourseStreams.ToListAsync();
     }
+    public async Task EnrollStudentAsync(int studentId, int courseStreamId)
+    {
+        var enrollment = new EnrolledStudent
+        {
+            StudentId = studentId,
+            CourseStreamId = courseStreamId
+        };
+
+        _context.EnrolledStudents.Add(enrollment);
+        await _context.SaveChangesAsync();
+    }
+
+    public async Task<bool> IsStudentEnrolledAsync(int studentId, int courseStreamId)
+    {
+        return await _context.EnrolledStudents
+            .AnyAsync(e => e.StudentId == studentId && e.CourseStreamId == courseStreamId);
```

Comment:
Flip this to `e.CourseStreamId == courseStreamId && e => e.StudentId == studentId`. In this case, you will receive better SQL and check the `course stream` first.

I think there always will be fewer courses than students

Reply:
Fixed